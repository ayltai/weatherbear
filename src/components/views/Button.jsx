import { IconButton, makeStyles, Tooltip, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const Button = props => {
    const classes = makeStyles(theme => ({
        container : {
            padding : theme.spacing(1),
        },
    }))();

    const { t, } = useTranslation();

    return (
        <Tooltip title={t(props.tooltip)}>
            <IconButton
                className={classes.container}
                edg={props.edge}
                size={props.size}
                onClick={() => {
                    if (props.onClick) props.onClick();
                }}>
                {props.icon}
            </IconButton>
        </Tooltip>
    );
};

Button.propTypes = {
    className : PropTypes.string,
    tooltip   : PropTypes.string,
    icon      : PropTypes.node,
    edge      : PropTypes.oneOf([
        'end',
        'start',
    ]),
    size      : PropTypes.oneOf([
        'small',
        'medium',
    ]),
    onClick   : PropTypes.func,
};
