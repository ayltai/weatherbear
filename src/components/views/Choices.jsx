import { List, ListItem, ListItemIcon, ListItemSecondaryAction, ListItemText, } from '@material-ui/core';
import { Delete, } from '@material-ui/icons';
import PropTypes from 'prop-types';
import React from 'react';
import { useTranslation, } from 'react-i18next';

import { Button, } from './Button';

export const Choices = props => {
    const { t, } = useTranslation();

    return (
        <List>
            {props.values.map((value, index) => (
                <ListItem
                    key={index}
                    button
                    onClick={() => {
                        if (props.onSelect) props.onSelect(value, index);
                    }}>
                    {props.value === value && (
                        <ListItemIcon>{props.icon}</ListItemIcon>
                    )}
                    <ListItemText
                        inset={props.value !== value}
                        primary={t(value)} />
                    <ListItemSecondaryAction>
                        <Button
                            tooltip='Delete'
                            edge='end'
                            icon={<Delete />}
                            onClick={() => {
                                if (props.onDelete) props.onDelete(value, index);
                            }} />
                    </ListItemSecondaryAction>
                </ListItem>
            ))}
        </List>
    );
};

Choices.propTypes = {
    className : PropTypes.string,
    icon      : PropTypes.node,
    value     : PropTypes.string,
    values    : PropTypes.arrayOf(PropTypes.string),
    onSelect  : PropTypes.func,
    onDelete  : PropTypes.func,
};
