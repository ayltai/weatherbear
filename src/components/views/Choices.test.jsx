import { ListItem, ListItemSecondaryAction, } from '@material-ui/core';
import { mount, shallow, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { Button, } from './Button';
import { Choices, } from './Choices';

const values = [ 'Line 1', 'Line 2', ];

describe('<Choices />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <Choices
                value={values[1]}
                values={values} />
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <Choices
                value={values[1]}
                values={values} />
        );
    });

    it('triggers onSelect when selected', () => {
        const handleSelect = jest.fn();
        const onSelect     = () => handleSelect();

        shallow(
            <Choices
                value={values[1]}
                values={values}
                onSelect={onSelect} />)
            .find(ListItem)
            .first()
            .simulate('click');

        expect(handleSelect.mock.calls.length).toEqual(1);
    });

    it('triggers onDelete when the delete icon is clicked', () => {
        const handleDelete = jest.fn();
        const onDelete     = () => handleDelete();

        shallow(
            <Choices
                value={values[1]}
                values={values}
                onDelete={onDelete} />)
            .find(ListItem)
            .first()
            .find(ListItemSecondaryAction)
            .find(Button)
            .simulate('click');

        expect(handleDelete.mock.calls.length).toEqual(1);
    });
});
