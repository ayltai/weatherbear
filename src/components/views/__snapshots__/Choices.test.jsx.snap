// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Choices /> renders correctly 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <li
    className="MuiListItem-container"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root MuiListItemText-inset"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Line 1
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </div>
    <div
      className="MuiListItemSecondaryAction-root"
    >
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root makeStyles-container-56"
        disabled={false}
        edg="end"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Delete"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
  <li
    className="MuiListItem-container"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      />
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Line 2
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </div>
    <div
      className="MuiListItemSecondaryAction-root"
    >
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root makeStyles-container-88"
        disabled={false}
        edg="end"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Delete"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</ul>
`;
